Source/dthread.cpp

```diff
@@ -78,95 +76,96 @@ void __fastcall dthread_send_delta(int pnum, char cmd, void *pbSrc, int dwLen)
 	SetEvent(sghWorkToDoEvent);
 	LeaveCriticalSection(&sgMemCrit);
 }
-// 679660: using guessed type char gbMaxPlayers;
 
 void __cdecl dthread_start()
 {
-	char *v0; // eax
-	char *v1; // eax
-
-	if (gbMaxPlayers != 1) {
-		sghWorkToDoEvent = CreateEvent(NULL, TRUE, FALSE, NULL);
-		if (!sghWorkToDoEvent) {
-			v0 = TraceLastError();
-			TermMsg("dthread:1\n%s", v0);
-		}
-		byte_52A508 = 1;
-		sghThread = (HANDLE)_beginthreadex(NULL, 0, dthread_handler, NULL, 0, &glpDThreadId);
-		if (sghThread == (HANDLE)-1) {
-			v1 = TraceLastError();
-			TermMsg("dthread2:\n%s", v1);
-		}
+	char *error_buf;
+
+	if (gbMaxPlayers == 1) {
+		return;
+	}
+
+	sghWorkToDoEvent = CreateEvent(NULL, TRUE, FALSE, NULL);
+	if (!sghWorkToDoEvent) {
+		error_buf = TraceLastError();
+		TermMsg("dthread:1\n%s", error_buf);
+	}
+
+	byte_52A508 = 1;

```

Comment:
Wish we had a name for `byte_52A508` but that can be done in follow-up PRs.

**Edit:** Perhaps `dthread_running`? And make it a boolean?

Reply:
Was thinking that my self, but unsure of what to name it.